apigen/src/main/java/com/distelli/graphql/apigen/STModel.java

```diff
@@ -319,7 +326,7 @@ private DataResolver toDataResolver(Type type) {
             TypeEntry typeEntry = referenceTypes.get(typeName);
             if ( !typeEntry.hasIdField() ) return null;
             DataResolver resolver = new DataResolver();
-            resolver.fieldType = typeName + ".Resolver";
+            resolver.fieldType = typeName + (isAsync ? ".AsyncResolver" : ".Resolver");
```

Comment:
Can we please change the code so it tries to `@Inject` both the AsyncResolver and non-async Resolver? We can then give priority to the async resolver. Then we can revert all these changes too.